tests/functional/test_wheel.py

```diff
@@ -214,7 +214,10 @@ def test_pip_wheel_with_pep518_build_reqs(script, data, common_wheels):
 @pytest.mark.network
 def test_pip_wheel_with_pep518_build_reqs_no_isolation(script, data,
                                                        common_wheels):
-    script.pip_install_local('-f', common_wheels, 'wheel', 'simplewheel==2.0')
+    script.pip_install_local(
+        '-f', common_wheels,
+        'setuptools>=40.2.0', 'wheel',
```

Comment:
I don't understand why this is necessary (setuptools>=40.2.0).

Reply:
Because without the version restriction, setuptools 39.1.0 (which is the one bundled by virtualenv, installed by default) is used and that version doesn't support PEP 517. With no isolation, it's the caller's responsibility to set up the build environment correctly, and under PEP 517 "correctly" includes "with a new enough version of setuptools".

Reply:
How come the test passes even with 39.1.0 then?

Reply:
AFAIK, the PEP 517 backend was added in 36.6.0.

Reply:
> How come the test passes even with 39.1.0 then?

It doesn't (now that projects with `pyproject.toml` go through PEP 517 processing). The backend implementation had a bug in it that needed fixing.

> AFAIK, the PEP 517 backend was added in 36.6.0.

Yes, but it had a Unicode handling bug till 40.2.0 (which was when the fix I submitted got released).

Reply:
I just ran it locally with no errors...

Reply:
I know, weird isn't it? Travis was consistently failing, but I was having problems reproducing the issue in *any* form. Are you running Windows or Unix? I'm on Windows so I put what I was seeing down to OS differences (I don't recall if I ever checked Appveyor, it's so slow running that it was nearly always out of date by the time it gave any results :-().

Regardless of strange "doesn't fail when it should" questions, I do think that explicitly installing a suitable version of setuptools is correct in this test - it shouldn't really be assuming that *any* version of setuptools is present in the local environment by default.

Reply:
I'm on Linux.